vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -995,4 +995,36 @@ public Style getStyle() {
                 "Dialog does not support adding styles to overlay");
     }
 
+    /**
+     * Close all the Dialog components associated within the current UI.
+     */
+    public static void closeAllDialogs() {
```

Comment:
Closing all dialogs right after opening one doesn't work:

```java
@Test
public void foobar() {
    Dialog dialog = new Dialog();
    dialog.open();
    
    Dialog.closeAllDialogs(); // Closing all dialogs right after opening doesn't work
    ui.getInternals().getStateTree().runExecutionsBeforeClientResponse();
    
    Assert.assertFalse(dialog.isOpened());
}
```

Reply:
The dialog is going to be really opened and attached only after runExecutionsBeforeClientResponse, that is the reason I used the particular IT to test it.

Reply:
This doesn't relate to how you're testing it. The same case fails with an IT just the same. Try this with an actual UI:

```java
button.addClickListener(event -> {
    dialog.open();
    Dialog.closeAllDialogs();
});
```

Result:

https://user-images.githubusercontent.com/1222264/210310840-56083aa7-7c4c-44a5-b315-5509e05ad410.mp4

Reply:
Good observation. It is a tricky case. As Dialog#open logic is delayed https://github.com/vaadin/flow-components/blob/master/vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java#L777 Thus in that particular case Dialog.closeAllDialogs() is executed first and Dialog is opened after that. So the order in code does not reflect the actual order.